### PR TITLE
Meltdown spectre 01

### PR DESCRIPTION
**Pull request by jbech-linaro**
@shovanuk @kylekirkby , first set of information to share publicly regarding the Meltdown and Spectre attacks. There will be updates to this page in the near future, but I think I'd like to get the current changes merged.

(I've also sneaked in the CVE number to two older issues, Mitre finally gave them to me).

// Joakim

### CODE REVIEW

**Comment by kylekirkby** [comment]
Thanks @jbech-linaro !

File: security-advisories/security-advisories.md [hunk 1]
```diff
@@ -18,6 +18,76 @@ If you have found a security issue in OP-TEE, please send us an email (see
...
+
+#### Current status:
+For `Armv8-A` builds we are typically running OP-TEE with Arm Trusted Firmware,
+for those there are patches currently being reviewed here:
+* https://github.com/ARM-software/arm-trusted-firmware/pull/1214
+
+For builds where we are not using Arm TF (typically `Armv7-A` builds) we have
+implemented mitigations that are being reviewed here:
+* https://github.com/OP-TEE/optee_os/pull/2047 (merged)
+* https://github.com/OP-TEE/optee_os/pull/2055
```
Comment (jforissier) [inline security-advisories/security-advisories.md]: This is slightly incorrect

File: security-advisories/security-advisories.md [hunk 2]
```diff
@@ -18,6 +18,76 @@ If you have found a security issue in OP-TEE, please send us an email (see
...
+as when doing syscalls from Trusted Applications. 
+
+#### Current status:
+Still under investigation, we have received GCC patches from Arm that can help us finding vulnerable areas.
+
+| Reported by  | CVE ID | OP-TEE ID | Affected versions |
+| ------------ |:------:| :-------: | ----------------- |
+| [Google Project Zero], [University of Pennsylvania], [University of Maryland], [Rambus], [Graz University of Technology], [University of Adelaide], [Data61] | [CVE-2017-5753](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-5753) | OP-TEE-2018-0001 | All versions prior to `OP-TEE 3.0.0` (not yet released) |
+
+### Variant 2: branch target injection (CVE-2017-5715)
```
Comment (jforissier) [inline security-advisories/security-advisories.md]: This section is slightly incomplete/incorrect, should probably differentiate between:
1. context switches from nsec to sec on the one hand,
2. secure world exceptions/syscalls on the other hand, causing transition from SEL0 to SEL1

Arm-TF 1214 and OP-TEE 2047 address (1) while OP-TEE 2055 (32-bits) and OP-TEE TBD (64-bits) address (2).

**Comment by jbech-linaro** [comment]
Fixed comments by Jerome. @shovanuk @kylekirkby , Ok by you? People are asking for this kind of information so I'd like to get this out as soon as possible. As mentioned in the previous comment, things are moving here, so I will probably add some more minor pull requests the comings days to this. But as of now we have the majority here.

**Pull request merged**